R/zchunk_LB171.LC_R_Cropland_Yh_GLU_irr.R

```diff
@@ -51,82 +51,48 @@ module_aglu_LB171.LC_R_Cropland_Yh_GLU_irr <- function(command, ...) {
-      L171.LC_bm2_R_irrHarvCropLand_C_Yh_GLU
-
-    L122.LC_bm2_R_HarvCropLand_C_Yh_GLU %>%
-      # Total cropland cover, will be used for where rainfed share is missing
-      rename(total = value) %>%
-      # Match the base year share by GCAM region, commodity, year and GLU.
-      left_join_error_no_match(L171.ag_irrHA_frac_R_C_GLU, by = c("GCAM_region_ID", "GCAM_commodity", "GLU")) %>%
-      # Calculate the rainfed cropland cover in all historical periods.
-      mutate(value = total * rfd_share,
-             # For the rainfed cropland table, missing values default to total cropland quantities; there are no missing currently (5/10/17).
-             value = if_else(is.na(value), total, value)) %>%
-      select(GCAM_region_ID, GCAM_commodity, GLU, year, value) ->
-      L171.LC_bm2_R_rfdHarvCropLand_C_Yh_GLU
-
-    # Extend irrigated cropland cover data to all years, 1700-2010
-    # Note: 1700, 1750, 1800, 1850, 1900, and 1950 are all zero; and this interpolation fills 1700-1970 with zeros, but no missing elsewhere.
-    L171.LC_bm2_R_irrHarvCropLand_C_Yh_GLU %>%
-      select(GCAM_region_ID, GCAM_commodity, GLU) %>%
-      unique() -> id_list
-    tibble(year = seq(1700, max(HISTORICAL_YEARS))) -> all_years
-    id_list %>%
-      repeat_add_columns(all_years) %>%
-      left_join(L171.LC_bm2_R_irrHarvCropLand_C_Yh_GLU, by = c("GCAM_region_ID", "GCAM_commodity", "GLU", "year")) %>%
-      group_by(GCAM_region_ID, GCAM_commodity, GLU) %>%
-      mutate(value = approx_fun(year, value)) %>%
-      ungroup() -> irrHarvCropLand_interp
-
-    # Extend rainfed and cropland cover data to all years, 1700-2010
-    # Note: 1700, 1750, 1800, 1850, 1900, and 1950 are all zero; and this interpolation fills 1700-1970 with zeros, but no missing elsewhere.
-    L171.LC_bm2_R_rfdHarvCropLand_C_Yh_GLU %>%
-      select(GCAM_region_ID, GCAM_commodity, GLU) %>%
-      unique() -> id_list
-    tibble(year = seq(1700, max(HISTORICAL_YEARS))) -> all_years
-    id_list %>%
-      repeat_add_columns(all_years) %>%
-      left_join(L171.LC_bm2_R_rfdHarvCropLand_C_Yh_GLU, by = c("GCAM_region_ID", "GCAM_commodity", "GLU", "year")) %>%
+    IrrRfdCropland <-
+        L122.LC_bm2_R_HarvCropLand_C_Yh_GLU %>%
+          left_join_error_no_match(L171.ag_irrHA_frac_R_C_GLU,
+                                   by = c("GCAM_region_ID", "GCAM_commodity",
+                                   "GLU")) %>%
+          mutate(irr.harvarea = value * irrHA_frac,
+                 irr.harvarea = replace(irr.harvarea, is.na(irr.harvarea), 0.0), # if no data, assume irrigated is zero
+                 rfd.harvarea = value * rfd_share,
+                 rfd.harvarea = replace(rfd.harvarea, is.na(rfd.harvarea), value)) # if no data, assume all rainfed
+
+    ## Extend to cover all years 1700-2010.
+    idvars <- c('GCAM_region_ID', 'GCAM_commodity', 'GLU', 'year')
+    allyr <- seq(min(HISTORICAL_YEARS), max(HISTORICAL_YEARS))
```

Comment:
It goes back to the first of the `HISTORICAL_YEARS`, whatever it happens to be.  It passes the time shift test because the problem was actually in the other table being joined; it needed to be filtered to years that were actually being used.